fields/evaluate.go

```diff
@@ -21,113 +21,191 @@ import (
 )
 
 const (
-	AggregateDefault AggregateMethod = iota
-	AggregateCount
-	AggregateMin
-	AggregateMax
-	AggregateSum
-	AggregateAvg
-	AggregateStateHistSeconds
-	AggregateStateHistPercent
-	AggregateStateHistRate
+	TimeAggregateDefault TimeAggregateMethod = iota
+	TimeAggregateCount
+	TimeAggregateMin
+	TimeAggregateMax
+	TimeAggregateSum
+	TimeAggregateAvg
+	TimeAggregateStateHistSeconds
+	TimeAggregateStateHistPercent
+	TimeAggregateStateHistRate
 )
 
-type AggregateMethod uint8
+type TimeAggregateMethod uint8
+
+const (
+	GroupAggregateDefault GroupAggregateMethod = iota
+	GroupAggregateCount
+	GroupAggregateMin
+	GroupAggregateMax
+	GroupAggregateSum
+	GroupAggregateAvg
+)
+
+type GroupAggregateMethod uint8
```

Comment:
Since we go with "aggregation" in the filed name, lets rename to `TimeAggreation...` and `GroupAggregation...`.

for the aggregation method type, let's drop the `Method` suffix  and go with
```
type TimeAggregation uint8
type GroupAggregation uint8
```

Otherwise, remove `StateHist` from the `TimeAggregation` names

Reply:
can do!